### PR TITLE
Document GraalVM usage with Log4j

### PR DESCRIPTION
**Pull request by ppkarwasz**
We document the usage of GraalVM together with all official Log4j API implementations.

Part of #2831.

### CODE REVIEW

File: src/site/antora/modules/ROOT/pages/graalvm.adoc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+Since version
+xref:release-notes.adoc#release-notes-2-25-0[`2.25.0`]
+both
+xref:manual/api.adoc[Log4j API]
+and
+xref:manual/implementation.adoc[its reference implementation]
+provide out-of-the-box support for creating native executables using
+https://www.graalvm.org/[GraalVM].
+
+Depending on the logging implementation you are using, additional steps might be required to compile a Log4j API application with GraalVM.
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/graalvm.adoc:29]: I am confused by the introduction and the sectioning following it. What is this page exactly documenting? Guiding applications using Log4j API to create GraalVM native images? What about applications using Log4j Core in combination with SLF4J?

Could you end the introduction paragraph with a dumb-proof _"This document guides you on ..."_ explanation, please?

File: src/site/antora/modules/ROOT/pages/graalvm.adoc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+Check our
+https://github.com/apache/logging-log4j-samples/tree/main/log4j-samples-graalvm[`log4j-samples-graalvm`]
+sample.
+====
+
+[#impl-simple]
+== Using Simple Logger
+
+If you use
+xref:manual/simple-logger.adoc[Simple Logger]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/graalvm.adoc:49]:
```suggestion
xref:manual/simple-logger.adoc[Simple Logger] shipped with xref:manual/api.html[Log4j API]
```

**Review by vy — CHANGES_REQUESTED** [review]
Liked the content, but there is room to make `graalvm.adoc` more dumb-proof, IMHO.

File: src/site/antora/modules/ROOT/pages/graalvm.adoc (hunk 1 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/graalvm.adoc]: I am documenting the usage of GraalVM with applications written against the Log4j API.
Honestly I didn't test the usage of third-party logging APIs, although I don't believe our bridges use any reflection, so they should be OK.

I modified the introduction in:
https://github.com/apache/logging-log4j2/pull/3054/commits/5031b1bd30b78271d3607b7456d22258376becdb

File: src/site/antora/modules/ROOT/pages/manual/installation.adoc
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/installation.adoc:1]: Fixed in https://github.com/apache/logging-log4j2/pull/3054/commits/5031b1bd30b78271d3607b7456d22258376becdb

File: src/site/antora/modules/ROOT/pages/manual/installation.adoc [hunk 3]
```diff
@@ -169,6 +169,30 @@ h| Gradle
...
 ====
 
+[#impl-simple]
+=== Installing Simple Logger
+
+The
+xref:manual/simple-logger.adoc[Simple Logger]
+implementation is embedded in the Log4j API and does not need any external dependency.
+
+Its usage is, however, opt-in and Log4j API will log an error to the
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/installation.adoc]: Not sure if the grammar dictates the subdivision into paragraphs, but you are right: this is certainly stylistically incorrect. Thanks for fixing it.

I did an additional rewording in https://github.com/apache/logging-log4j2/pull/3054/commits/7f71473a2e95cb6c9c2ccca156414026736bcdde

File: src/site/antora/modules/ROOT/pages/graalvm.adoc [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+Check out
+the https://github.com/apache/logging-log4j-samples/tree/main/log4j-samples-graalvm[`log4j-samples-graalvm`]
+project.
+====
+
+[#impl-simple]
+== Using Simple Logger
+
+If you use
+xref:manual/simple-logger.adoc[Simple Logger] shipped with xref:manual/api.html[Log4j API]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/graalvm.adoc:52]:
```suggestion
xref:manual/simple-logger.adoc[Simple Logger] shipped with xref:manual/api.adoc[Log4j API]
```

**Pull request merged**